### PR TITLE
Upgrade Gradle and gradle wrappers.

### PR DESCRIPTION
**Pull request by prideout**
Some of our custom gradle scripting stopped working after allowing Android Studio to upgrade, this fixes those issues.

### CODE REVIEW

File: android/build/filament-tasks.gradle [hunk 1]
```diff
@@ -64,8 +64,8 @@ class MaterialCompiler extends DefaultTask {
     File matcPath
 
     @SuppressWarnings("GroovyUnusedDeclaration")
-    @InputDirectory
-    File inputDir
+    @InputFile
```
Comment (romainguy) [inline android/build/filament-tasks.gradle]: Are we sure about this change?

File: android/build/filament-tasks.gradle [hunk 2]
```diff
@@ -64,8 +64,8 @@ class MaterialCompiler extends DefaultTask {
...
+    File inputFile
```
Comment (romainguy) [inline android/build/filament-tasks.gradle]: The field is renamed, doesn't this break the script?

File: android/samples/image-based-lighting/app/build.gradle [hunk 3]
```diff
@@ -8,7 +8,7 @@ compileMaterials {
     group 'Filament'
     description 'Compile materials'
 
-    inputDir = file("src/main/materials")
+    inputFile = file("src/main/materials/clear_coat.mat")
```
Comment (romainguy) [inline android/samples/image-based-lighting/app/build.gradle]: Hmm I explicitly used a directory to compile ALL materials.

**Comment by prideout** [comment]
Yeah the directory-based approach no longer works with newer gradle, and
none of our samples have more than one material file anyway.  Maybe we
could keep the directory feature if we add an explicit glob?

**Comment by prideout** [comment]
To be more specific, new gradle passes the raw directory path straight to
matc, rather than passing each of its files to matc.

File: android/build/filament-tasks.gradle (hunk 2 above)
Comment (prideout) [reply android/build/filament-tasks.gradle]: The field name is not directly referenced in the script, I think this works because of the annotation.

**Comment by pixelflinger** [comment]
@prideout and this "new" behavior is not a bug? They just broke stuff?

**Comment by prideout** [comment]
Old gradle passed only the contents of out-of-date directories, newer gradle passes both the contents and the directory itself.  Since they bumped their major version number, they technically did not break anything.

A better fix than what I've done is to keep Romain's directory-based approach, but add a one-liner to the custom task that skips over directory paths.  Stand by, I will amend the PR.

File: android/build/filament-tasks.gradle (hunk 1 above)
Comment (prideout) [reply android/build/filament-tasks.gradle]: Restored this, replaced with a different fix.

File: android/samples/image-based-lighting/app/build.gradle (hunk 3 above)
Comment (prideout) [reply android/samples/image-based-lighting/app/build.gradle]: This functionality has been restored in the latest version of my fix.

**Pull request merged**